src/main/scala/mesosphere/marathon/core/task/tracker/InstanceTracker.scala

```diff
@@ -50,6 +50,12 @@ trait InstanceTracker {
   def updateStatus(instance: Instance, mesosStatus: mesos.Protos.TaskStatus, updateTime: Timestamp): Future[Done]
 
   def reservationTimeout(instanceId: Instance.Id): Future[Done]
+
+  def goalDecommissioned(instanceId: Instance.Id): Future[Done]
```

Comment:
Seeing the `updateStatus` method above, it seems more warranted that we introduce a single `updateGoal` method with the enum. Eager to hear your thoughts on this.